pkg/asset/cluster/azure/azure.go

```diff
@@ -30,13 +31,50 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 // PreTerraform performs any infrastructure initialization which must
 // happen before Terraform creates the remaining infrastructure.
 func PreTerraform(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig) error {
-	if len(installConfig.Config.Azure.ResourceGroupName) == 0 {
+	session, err := installConfig.Azure.Session()
+	if err != nil {
+		return fmt.Errorf("failed to get session: %w", err)
+	}
+
+	if err := tagResourceGroup(ctx, clusterID, installConfig, session); err != nil {
+		return err
+	}
+
+	if err := tagVNet(ctx, clusterID, installConfig, session); err != nil {
+		return err
+	}
+	return nil
+}
+
+func tagVNet(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig, session *icazure.Session) error {
+	if len(installConfig.Config.Azure.VirtualNetwork) == 0 {
 		return nil
 	}
 
-	session, err := installConfig.Azure.Session()
+	resourceGroupName := installConfig.Config.Azure.NetworkResourceGroupName
+
+	vnetClient, err := armnetwork.NewVirtualNetworksClient(session.Credentials.SubscriptionID, session.TokenCreds, nil)
 	if err != nil {
-		return errors.Wrap(err, "failed to get session")
+		return fmt.Errorf("failed to get the virtual network client: %w", err)
+	}
+
+	tagKey, tagValue := sharedTag(clusterID)
+	tags := armnetwork.TagsObject{
+		Tags: map[string]*string{tagKey: tagValue},
+	}
+
+	if _, err := vnetClient.UpdateTags(
```

Comment:
I changed this to use `UpdateTags` and asked QE to confirm whether the call overwrite existing tags in the vnet.

Reply:
Can't you just create a vnet and run the `UpdateTags` in a standalone program? It'll be faster to validate/experiment than waiting on QE to deploy a cluster to test it.

Reply:
Sure I'll give that a go.

Reply:
that was a good idea - it did override existing tags - fixed the code so that doesn't happen.